pkg/chunked/storage_linux.go

```diff
@@ -994,6 +1023,35 @@ func closeDestinationFiles(files chan *destinationFile, errors chan error) {
 	close(errors)
 }
 
+func (c *chunkedDiffer) recordFsVerity(path string, roFile *os.File) error {
+	if c.useFsVerity == graphdriver.DifferFsVerityDisabled {
+		return nil
+	}
+
+	err := fsverity.EnableVerity(path, int(roFile.Fd()))
+	if err != nil {
+		if c.useFsVerity == graphdriver.DifferFsVerityRequired {
```

Comment:
This should handle EEXIST for the fs-verity already enabled case and make sure to then read and record the existing digest.

Reply:
`fsverity.EnableVerity` returns nil if fs-verity was already enabled, so we should already get into the `fsverity.MeasureVerity` function even if `DifferFsVerityRequired`